sqlx-postgres/src/types/ts_vector.rs

```diff
@@ -0,0 +1,286 @@
+                    .map(|LexemeMeta { position, weight }| {
+                        match weight {
+                            3 => format!("{position}A"),
+                            2 => format!("{position}B"),
+                            1 => format!("{position}C"),
+                            // 'D' is the default value and does not need to be displayed
+                            _ => format!("{position}"),
+                        }
+                    })
+                    .collect::<Vec<_>>()
+                    .join(",");
+
+                f.write_str(&format!("'{}':{}", word, position))?;
+            }
+
+            if words.peek().is_some() {
+                f.write_char(' ')?;
+            }
+        }
+
+        Ok(())
+    }
+}
+
+impl TryFrom<&[u8]> for TsVector {
+    type Error = BoxDynError;
+
+    /// Decode binary data into [`TsVector`] based on the binary data format defined in
+    /// https://github.com/postgres/postgres/blob/252dcb32397f64a5e1ceac05b29a271ab19aa960/src/backend/utils/adt/tsvector.c#L399
+    fn try_from(bytes: &[u8]) -> Result<Self, Self::Error> {
+        let mut reader = Cursor::new(bytes);
+        let mut words = vec![];
+
+        let num_lexemes = reader.read_u32::<BigEndian>()?;
+
+        for _ in 0..num_lexemes {
+            let mut lexeme = vec![];
+
+            reader.read_until(b'\0', &mut lexeme)?;
+
+            let num_positions = reader.read_u16::<BigEndian>()?;
+            let mut positions = Vec::<LexemeMeta>::with_capacity(num_positions as usize);
+
+            if num_positions > 0 {
+                for _ in 0..num_positions {
+                    let position = reader.read_u16::<BigEndian>()?;
+                    positions.push(LexemeMeta::from(position));
+                }
+            }
+
+            words.push(Lexeme {
+                word: str::from_utf8(&lexeme)?.trim_end_matches('\0').to_string(),
+                positions,
+            });
+        }
+
+        Ok(Self { words })
+    }
+}
+
+impl TryInto<Vec<u8>> for &TsVector {
+    type Error = BoxDynError;
+
+    fn try_into(self) -> Result<Vec<u8>, Self::Error> {
+        let buf: &mut Vec<u8> = &mut vec![];
+
+        buf.write_u32::<BigEndian>(u32::try_from(self.words.len())?)?;
+
+        for lexeme in &self.words {
+            buf.write(lexeme.word.as_bytes())?;
+            buf.write(&[b'\0'])?;
+
+            buf.write_u16::<BigEndian>(u16::try_from(lexeme.positions.len())?)?;
+
+            if !lexeme.positions.is_empty() {
+                for lexeme_meta in &lexeme.positions {
+                    buf.write_u16::<BigEndian>(lexeme_meta.into())?;
+                }
+            }
+        }
+
+        buf.flush()?;
+
+        Ok(buf.to_vec())
+    }
+}
+
+impl FromStr for TsVector {
+    type Err = ParseLexemeMetaError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let mut words: Vec<Lexeme> = vec![];
+
+        for word in s.split(' ') {
```

Comment:
```rust
let row = pool.fetch_one(r#"SELECT $$'    A'$$::tsvector;"#).await?;
assert_eq!(row.get::<TsVector, _>(0).to_string(), "'    A'");
```
#### Error
```
assertion `left == right` failed
  left: "'' '' '' '' 'A'"
 right: "'    A'"
```

Reply:
Other than that, I think `TsVector` is perfect, thanks for your work 👍!

Reply:
Thanks @wyhaya. This has been fixed (and a test has been added)